server/controllers/stock/index.js

```diff
@@ -129,78 +130,205 @@ function createStock(req, res, next) {
+    const allLots = await core.getLotsDepot(null, { depot_uuid : movement.depot_uuid });
+
+    // reverse all selected lots
+    const availableLots = allLots.filter(lot => {
+      return inventoryLotUuids.includes(lot.uuid);
+    });
+
+    // pass reverse operations
+    const trx = db.transaction();
+
+    const positiveAdjustmentUuid = uuid();
+    const negativeAdjustmentUuid = uuid();
+    const positiveQuantities = availableLots.filter(lot => lot.quantity > 0);
+    const negativeQuantities = availableLots.filter(lot => lot.quantity < 0);
+
+    positiveQuantities.forEach(lot => {
+      const reverseMovementObject = {
+        uuid : db.bid(uuid()),
+        lot_uuid : db.bid(lot.uuid),
+        depot_uuid : db.bid(movement.depot_uuid),
+        document_uuid : db.bid(negativeAdjustmentUuid),
+        quantity : lot.quantity,
+        unit_cost : lot.unit_cost,
+        date : new Date(movement.date),
+        entity_uuid : movement.entity_uuid,
+        is_exit : 1,
+        flux_id : core.flux.INVENTORY_RESET,
+        description : movement.description,
+        user_id : req.session.user.id,
+        period_id : periodId,
+      };
+      trx.addQuery('INSERT INTO stock_movement SET ?', reverseMovementObject);
+    });
+
+    negativeQuantities.forEach(lot => {
+      const reverseMovementObject = {
+        uuid : db.bid(uuid()),
+        lot_uuid : db.bid(lot.uuid),
+        depot_uuid : db.bid(movement.depot_uuid),
+        document_uuid : db.bid(positiveAdjustmentUuid),
+        quantity : lot.quantity,
+        unit_cost : lot.unit_cost,
+        date : new Date(movement.date),
+        entity_uuid : movement.entity_uuid,
+        is_exit : 0,
+        flux_id : core.flux.INVENTORY_RESET,
+        description : movement.description,
+        user_id : req.session.user.id,
+        period_id : periodId,
+      };
+      trx.addQuery('INSERT INTO stock_movement SET ?', reverseMovementObject);
+    });
+
+    const negativeAdjustmentParams = [
+      db.bid(negativeAdjustmentUuid), 1, req.session.project.id, req.session.enterprise.currency_id,
+    ];
+
+    const positiveAdjustmentParams = [
+      db.bid(positiveAdjustmentUuid), 0, req.session.project.id, req.session.enterprise.currency_id,
+    ];
+
+    if (req.session.enterprise.settings.enable_auto_stock_accounting) {
+      if (positiveQuantities.length > 0) {
+        trx.addQuery('CALL PostStockMovement(?)', [negativeAdjustmentParams]);
+      }
+
+      if (negativeQuantities.length > 0) {
+        trx.addQuery('CALL PostStockMovement(?)', [positiveAdjustmentParams]);
+      }
+    }
+
+    // reset all previous lots
+    await trx.execute();
+
+    // pass inventory adjustment as new movement
+    const document = {
+      uuid : uuid(),
+      date : new Date(movement.date),
+      user : req.session.user.id,
+    };
+    const positiveLots = lots
+      .filter(lot => lot.quantity > 0 && (lot.quantity !== lot.oldQuantity))
```

Comment:
I think you already did the ` (lot.quantity !== lot.oldQuantity)` check once, right?